lib/transloadit/assembly.rb

```diff
@@ -48,17 +48,22 @@ def create!(*ios)
     params[:steps] = _wrap_steps_in_hash(params[:steps]) if !params[:steps].nil?
 
     extra_params = {}
-    # update the payload with file entries
-    ios.each_with_index {|f, i| extra_params.update :"file_#{i}" => f }
-
     extra_params.merge!(self.options[:fields]) if self.options[:fields]
 
-    _do_request(
-      '/assemblies',
-      params,
-      'post',
-      extra_params
-    ).extend!(Transloadit::Response::Assembly)
+    # retry 2 more times on rate limit response.
+    trials = 3
+    (1..trials).each do |trial|
+      # update the payload with file entries
+      ios.each_with_index {|f, i| extra_params.update :"file_#{i}" => f }
```

Comment:
After each request trial, the upload file streams are closed. This line needs to run for every loop cycle because it reloads the payload with the reopened file streams. The file streams are reopened [here](https://github.com/transloadit/ruby-sdk/pull/33#discussion-diff-85099638R174)